contrib/docker-integration/README.md

```diff
@@ -1,136 +1,63 @@
-```
-
-### Set /etc/hosts entry
-Find the non-localhost ip address of local machine
-
-### Run bats
-Run the bats tests after updating /etc/hosts, installing the certificates, and
-running the `docker-compose` script.
-```
-bats -p .
-```
-
-## Configurations
-
-Port | V2 | TLS | Authentication
---- | --- | --- | ---
-5000 | yes | no | none
-5002 | yes | no | none
-5440 | yes | yes | none
-5441 | yes | yes | basic (testuser/passpassword)
-5442 | yes | yes | TLS client
-5443 | yes | yes | TLS client (no CA)
-5444 | yes | yes | TLS client + basic (testuser/passpassword)
-5445 | yes | yes (no CA) | none
-5446 | yes | yes (no CA) | basic (testuser/passpassword)
-5447 | yes | yes (no CA) | TLS client
-5448 | yes | yes (SSLv3) | none
+These integration tests cover interactions between registry clients such as
+the docker daemon and the registry server. All tests can be run using the
+[golem integration test runner](https://github.com/docker/golem)
+
+The integration tests configure components using docker compose
+(see docker-compose.yaml) and the runner can be using the golem
+configuration file (see golem.conf).
+
+## Running integration tests
+
+### Run using multiversion script
+
+The integration tests in the `contrib/docker-integration` directory can be simply
+run be executing the run script `./run_multiversion.sh`. If there is no running
+daemon to connect to, run as `./run_multiversion.sh -d`.
+
+This command will build the distribution image from the locally checked out
+version and run against multiple versions of docker defined in the script. To
+run a specific version of the registry or docker, Golem will need to be
+executed manually.
+
+### Run manually using Golem
+
+Using the golem tool directly allows running against multiple versions of
+the registry and docker. Running against multiple versions of the registry
+can be useful for testing changes in the docker daemon which are not
+covered by the default run script.
+
+#### Installing Golem
+
+Golem is distributed as an executable binary which can be installed from
+the [release page](https://github.com/docker/golem/releases/tag/v0.1).
+
+#### Running golem with docker
+
+Additionally golem can be run as a docker image requiring no additonal
+installation.
+
+`docker run --privileged -v "$GOPATH/src/github.com/docker/distribution/contrib/docker-integration:/test" -w /test distribution/golem golem -rundaemon .`
+
+#### Golem custom images
+
+Golem tests version of software by defining the docker image to test.
+
+Run with registry 2.2.1 and docker 1.10.3
+
+`golem -i golem-dind:latest,docker:1.10.3-dind,1.10.3 -i golem-distribution:latest,registry:2.2.1 .`
+
+
+#### Use golem caching for developing tests
+
+Golem allows caching image configuration to reduce test start up time.
+Using this cache will allow tests with the same set of images to start
+up quickly. This can be useful when developing tests and needing the
+test to run quickly. If changes are which effect the image (such as
```

Comment:
Updated to "If there are changes", removing the "are" was not enough to make the sentence correct